### PR TITLE
fix(rich-text-editor): [rich-text-editor] fix data binding doen't work

### PR DESCRIPTION
**Pull request by gweesin**
# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-vue/blob/dev/CONTRIBUTING.md)
- [] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #1714 

## What is the new behavior?

RichTextEditor data binding should be worked now.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `<textarea>` alongside the rich text editor for plain text editing.
	- Enhanced the rich text editor's responsiveness to model value changes for improved interactivity.

### CODE REVIEW

File: packages/renderless/package.json [hunk 1]
```diff
@@ -40,6 +40,7 @@
     "xss": "1.0.11"
   },
   "devDependencies": {
+    "@tiptap/core": "~2.1.16",
```
Comment (kagol) [inline packages/renderless/package.json]: @gweesin renderless里不要引入第三方依赖哈~应该通过模板setup函数的参数`extendOptions`进行传递，供renderless中的rich-text-editor逻辑使用，而不是在renderless中的rich-text-editor中直接安装和引入。

示例代码如下：

组件模板中传入：
```
setup(props, context) {
    return setup({
      props,
      context,
      renderless,
      api,
      mono: true,
      extendOptions: {
        TinyTiptap,
        ...
      }
    })
}
```

renderless中引入：
```
export const renderless = (
  props,
  { computed, onBeforeUnmount, reactive },
  { vm, emit },
  {
    TinyTiptap,
    ...
  }
) => {
  ...
}
```

File: packages/renderless/src/rich-text-editor/vue.ts [hunk 2]
```diff
@@ -10,6 +10,8 @@ import {
   eventClick,
   Active
 } from './index'
+import type { ISharedRenderlessParamHooks } from '@/types'
+import type { Editor as TipTapEditor } from '@tiptap/core'
```
Comment (kagol) [inline packages/renderless/src/rich-text-editor/vue.ts]: 这里参考上一条检视意见

File: packages/renderless/package.json (hunk 1 above)
Comment (gweesin) [reply packages/renderless/package.json]: 这里实际上已经是 Tiptap 的对象，引入主要是为了处理类型没有传递过来的问题，如果去掉的话，那类型就丢失了，后续是否需要考虑类型传递问题以获得更好的编码体验？

**Review by kagol — APPROVED** [review]

File: packages/renderless/package.json (hunk 1 above)
Comment (kagol) [reply packages/renderless/package.json]: @gweesin 类型的传递也是一样的哈~从模板传递过来

Comment (kagol) [reply packages/renderless/package.json]: 懂你的意思了

**Pull request merged**